dawn/src/dawn/IIR/StencilFunctionInstantiation.cpp

```diff
@@ -392,8 +393,15 @@ void StencilFunctionInstantiation::update() {
          !metadata_.isAccessType(FieldAccessType::Field, AccessID))
         continue;
 
-      AccessUtils::recordWriteAccess(inputOutputFields, inputFields, outputFields, AccessID,
-                                     std::optional<Extents>(), interval_);
+      AccessUtils::recordWriteAccess(
+          inputOutputFields, inputFields, outputFields, AccessID, std::optional<Extents>(),
+          interval_,
+          metadata_.isAccessType(FieldAccessType::Field, AccessID)
+              ? metadata_.getFieldDimensions(AccessID)
+              : sir::FieldDimensions(
+                    sir::HorizontalFieldDimension(ast::cartesian, {true, true}),
```

Comment:
I tried to explain in detail this hack [here](https://stackoverflow.com/c/dawn/questions/32).
In short: 

- if the access id is found in the metadata it means that it is an actual `Field` registered in the stencil (can be API or temporary), so the dimensions can be retrieved from the metadata,

- otherwise it means it's an access id forged to represent a stencil function call (nested as argument in this stencil function call) => it's not a field => dimensions don't make much sense probably ... since this dimensions are never retrieved, here we just set arbitrary ones (that's the hack).

Conclusion is that `StencilFunctionInstantiation` needs a refactoring.

Why should we want to extract a variable here?

Reply:
> Why should we want to extract a variable here?

Because you can give it a name.

Reply:
Sure can do. Just thought that since it's a hack we want it gone as soon as possible. Also the access id is not the same for the other cases below, so the variable cannot be reused.